powerline/bindings/lemonbar/powerline-lemonbar.py

```diff
@@ -0,0 +1,199 @@
+					},
+				))
+
+				if not self.run_once and self.common_config['reload_config']:
+					interval = self.common_config['interval']
+					self.config_loader.set_interval(interval)
+					self.run_loader_update = (interval is None)
+					if interval is not None and not self.config_loader.is_alive():
+						self.config_loader.start()
+
+			self.ext_config = config['ext'][self.ext]
+
+			top_theme = (
+				self.ext_config.get('top_theme')
+				or self.common_config['default_top_theme']
+			)
+			self.theme_levels = (
+				os.path.join('themes', top_theme),
+				os.path.join('themes', self.ext, '__main__'),
+			)
+			self.renderer_options['theme_kwargs']['top_theme'] = top_theme
+
+			if self.ext_config != self.prev_ext_config:
+				ext_config_differs = True
+				if (
+					not self.prev_ext_config
+					or self.ext_config.get('components') != self.prev_ext_config.get('components')
+				):
+					self.setup_components(self.ext_config.get('components'))
+				if (
+					not self.prev_ext_config
+					or self.ext_config.get('local_themes') != self.prev_ext_config.get('local_themes')
+				):
+					self.renderer_options['local_themes'] = self.get_local_themes(self.ext_config.get('local_themes'))
+				load_colorscheme = (
+					load_colorscheme
+					or not self.prev_ext_config
+					or self.prev_ext_config['colorscheme'] != self.ext_config['colorscheme']
+				)
+				load_theme = (
+					load_theme
+					or not self.prev_ext_config
+					or self.prev_ext_config['theme'] != self.ext_config['theme']
+				)
+				self.prev_ext_config = self.ext_config
+
+		create_renderer = load_colors or load_colorscheme or load_theme or common_config_differs or ext_config_differs
+
+		if load_colors:
+			self._purge_configs('colors')
+			self.colors_config = self.load_colors_config()
+
+		if load_colorscheme or load_colors:
+			self._purge_configs('colorscheme')
+			if load_colorscheme:
+				self.colorscheme_config = self.load_colorscheme_config(self.ext_config['colorscheme'])
+			self.renderer_options['theme_kwargs']['colorscheme'] = (
+				Colorscheme(self.colorscheme_config, self.colors_config))
+
+		if load_theme:
+			self._purge_configs('theme')
+			self.renderer_options['theme_config'] = self.load_theme_config(self.theme or self.ext_config.get('theme', 'default'))
+
+		if create_renderer:
+			Renderer = self.get_module_attr(self.renderer_module, 'renderer')
+			if not Renderer:
+				if hasattr(self, 'renderer'):
+					return
+				else:
+					raise ImportError('Failed to obtain renderer')
+
+			# Renderer updates configuration file via segments’ .startup thus it
+			# should be locked to prevent state when configuration was updated,
+			# but .render still uses old renderer.
+			try:
+				renderer = Renderer(**self.renderer_options)
+			except Exception as e:
+				self.exception('Failed to construct renderer object: {0}', str(e))
+				if not hasattr(self, 'renderer'):
+					raise
+			else:
+				self.renderer = renderer
+
+if __name__ == '__main__':
```

Comment:
I would highly suggest moving *everything*, but a single function `main` out to `powerline/commands/lemonbar.py`. Reasoning: this makes it easy to create a compatibility script `powerline-bar.py`. (Symlinks and use of `subprocess.Popen` or equivalents in compatibility script is not that good idea because most files in `bindings` directory are supposed to be movable.)

Since arguments definition is getting bolder it should be in `powerline.commands.lemonbar.get_argparser`, similar to `powerline.commands.main` for the same reason: make it possible to create documentation and man page if needed.